### PR TITLE
Attempt to fix broken compilation on some systems caused by #1047

### LINKED ISSUE #1064: plugin needed to handle lto object following latest pulls
Hi @radfish and @fluffypony, I've just pulled down the latest merges and started to build on my system but it's starting to fail here:

```
[  9%] Linking CXX static library libotshell_utils.a
/usr/bin/ar: CMakeFiles/otshell_utils.dir/windows_stream.cpp.o: plugin needed to handle lto object
/usr/bin/ar: CMakeFiles/otshell_utils.dir/runoptions.cpp.o: plugin needed to handle lto object
/usr/bin/ar: CMakeFiles/otshell_utils.dir/utils.cpp.o: plugin needed to handle lto object
/usr/bin/ar: CMakeFiles/otshell_utils.dir/ccolor.cpp.o: plugin needed to handle lto object
/usr/bin/ranlib: windows_stream.cpp.o: plugin needed to handle lto object
/usr/bin/ranlib: runoptions.cpp.o: plugin needed to handle lto object
/usr/bin/ranlib: utils.cpp.o: plugin needed to handle lto object
/usr/bin/ranlib: ccolor.cpp.o: plugin needed to handle lot object
```

This error with /ar and /ranlib appears throughout the rest of the compile with the following libraries:

```libcrypto.a```,  ```libcommon.a```, ```libringct.a```, ```libblocks.a```, ```libblockchain_db.a```, ```libcryptonote_core.a```, ```libcryptonote_protocol.a```, ```librpc.a```, ```libwallet.a```, ```libp2p.a```

And then compilation finally fails when trying to link monero-wallet-cli

I found this if it helps? https://stackoverflow.com/questions/39236917/using-gccs-link-time-optimization-with-static-linked-libraries

Which makes me wonder whether PR #1047 possibly introduced this breaking change?

### PR DESCRIPTION
**Pull request by iDunk5400**
Since #1047 was merged, my Ubuntu 14.04 build environment consisting of four GCC compiler versions (4.8.5, 4.9.4, 5.4.1 and 6.2.0), all managed by update-alternatives, is able to compile monero only with GCC 4.8.5. After switching to any other compiler, there is a massive amount of unresolved dependencies when linking.

This is an attempt to preserve the fix to #1026 while not breaking build environments not affected by that issue. This may also fix #1064.

### CODE REVIEW

**Comment by NanoAkron** [comment]
Fingers crossed :)

**Comment by NanoAkron** [comment]
Hi @iDunk5400, I can report that this successfully fixes #1064.

@alown - do you know if this still fixes your issue in #1026?

**Comment by danrmiller** [comment]
I had the issue #1064 problem with ubuntu 16.04 amd64. PR #1065 fixes it for me.

**Comment by alown** [comment]
@NanoAkron, yes this seems to have fixed #1026.

**Comment by iDunk5400** [comment]
Thanks for your feedback, guys.

**Comment by moneroexamples** [comment]
It fixies the compilation issues on 16.04 for me as well. thx

File: CMakeLists.txt [hunk 1]
```diff
@@ -502,12 +502,15 @@ else()
       # -DCHOST=... or -DCMAKE_AR=... and -DCMAKE_RANLIB=...
```
Comment (radfish) [inline CMakeLists.txt]: This comment needs fixing now: to "... specify -DCHOST and optionally -DCMAKE_AR and -DCMAKE_RANLIB if they need to be different from ${CHOST}-gcc{ar,ranlib}"

**Comment by radfish** [comment]
Ok, sorry about that, so judging from this patch, update-alternatives is causing CMAKE_AR, CMAKE_RANLIB to be set to something other than gcc-ar and gcc-ranlib, correct?

**Comment by radfish** [comment]
Oh, just read [1]. So, the problem is that CMAKE_AR is always defined (to /usr/bin/ar by default), so we should probably not ever check whether it is defined or not. How about remove the lines `if(NOT DEFINED CMAKE_AR)` and `if(NOT DEFINED CMAKE_RANLIB)` and support modifying them only by passing `-DCHOST` (without supporting passing -DCMAKE_AR and -DCMAKE_RANLIB), and update that comment accordingly?

[1] https://stackoverflow.com/questions/39236917/using-gccs-link-time-optimization-with-static-linked-libraries

**Pull request merged**